server-spi-private/src/main/java/org/keycloak/connections/httpclient/HttpClientProvider.java

```diff
@@ -53,8 +52,8 @@ public interface HttpClientProvider extends Provider {
      * Helper method
      *
      * @param uri
-     * @return response stream, you must close this stream or leaks will happen
+     * @return Body of the response
      * @throws IOException On network errors, no content being returned or a non-2xx HTTP status code
      */
-    public InputStream get(String uri) throws IOException;
+    public String get(String uri) throws IOException;
```

Comment:
This means that every HTTP response returns a `String`. Is this really the case? Considering extensions that use this (unsupported) API, I'd suggest considering addition of a new method (e.g. `getAsString`, perhaps with a default implementation) rather than changing existing method.

Reply:
All occurrences I found in the code fetch either JSON or XML from a remote source. As stated in the original issue, and handling of encoding is lost in those cases. 

Extensions can still use `getHttpClient()` if they want to go beyond this. As it is an internal API, I'd rather keep its surface small. 

If that doesn't convince you, I'd go for two methods `getString()` and `getInputStream()` with JavaDoc explaining when to use one or the other.

Reply:
I like the new way where the encoding is correctly handled, so the `getString` method is a valuable addition.

Since this provider can be used to retrieve binary data in simple use cases (e.g. retrieve image in a mapper), a `HttpClient` reference is viable but adds unnecesary complexity. Thus the two methods would be my preferred option.